src/main/scala/edu/ie3/simona/scheduler/SchedulerHelper.scala

```diff
@@ -452,21 +446,31 @@ trait SchedulerHelper extends SimonaActorLogging {
       context.unwatch(trig.agent)
     )
 
-    /* notify listeners */
-    notifyListener(
-      Done(
-        endTick,
-        totalSimDuration,
-        stateData.runtime.noOfFailedPF,
-        errorInSim
-      )
-    )
-
     /* notify start sender */
     if (errorInSim) {
       stateData.runtime.initSender ! SimulationFailureMessage
+
+      /* notify listeners */
+      notifyListener(
+        Done(
+          stateData.time.nowInTicks,
+          totalSimDuration,
+          stateData.runtime.noOfFailedPF,
+          errorInSim
+        )
+      )
     } else {
       stateData.runtime.initSender ! SimulationSuccessfulMessage
+
+      /* notify listeners */
+      notifyListener(
```

Comment:
Can't you use this in both cases?

Reply:
I think since the code in 452-460 and in 464-472 are the same, you can just place it above the if-switch, like it was before.

Maybe also add a comment that the usage of `min` is necessary because the scheduler overshoots the target tick by 1 at the end of the simulation.